jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java

```diff
@@ -102,6 +103,15 @@ protected void handleNode(final Node node, final Element element, final String u
                 break;
             } else if ("signalEventDefinition".equals(nodeName)) {
                 String type = ((Element) xmlNode).getAttribute("signalRef");
+                Map<String, Signal> signals = (Map<String, Signal>) ((ProcessBuildData) parser.getData()).getMetaData("Signals");
+                
+                if (signals != null && signals.containsKey(type)) {
+                    Signal signal = signals.get(type);                      
+                    type = signal.getName();
+                    if (type == null) {
+                        throw new IllegalArgumentException("Signal definition must have a name attribute");
```

Comment:
Hello Marco,

I took this code change from these commits [1][2] from branch 6.3.x and the
change was in the IntermediateThrowEventHandler class. Am I missing
something?



[1]
https://github.com/droolsjbpm/jbpm/commit/1098074db5d0e39830267669970b9160e903c332
[2]
https://github.com/droolsjbpm/jbpm/commit/cadeff2c15305c4c6ce2b0c539f876cbb6a1913a

--
*William Antônio Siqueira*
*Java Support Analyst*

*http://fxapps.blogspot.com <http://fxapps.blogspot.com>*
*http://www.williamantonio.wordpress.com
<http://www.williamantonio.wordpress.com>*
*http://williamprogrammer.com <http://williamprogrammer.com>*

2015-11-23 8:35 GMT-02:00 Marco Rietveld <notifications@github.com>:

> In jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java
> <https://github.com/droolsjbpm/jbpm/pull/331#discussion_r45588888>:
>
> > @@ -102,6 +103,15 @@ protected void handleNode(final Node node, final Element element, final String u
> >                  break;
> >              } else if ("signalEventDefinition".equals(nodeName)) {
> >                  String type = ((Element) xmlNode).getAttribute("signalRef");
> > +                Map<String, Signal> signals = (Map<String, Signal>) ((ProcessBuildData) parser.getData()).getMetaData("Signals");
> > +
> > +                if (signals != null && signals.containsKey(type)) {
> > +                    Signal signal = signals.get(type);
> > +                    type = signal.getName();
> > +                    if (type == null) {
> > +                        throw new IllegalArgumentException("Signal definition must have a name attribute");
>
> Please move to the RuleFlowProcessValidator class, thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/droolsjbpm/jbpm/pull/331/files#r45588888>.
>